### PR TITLE
feat(j-s): Improve case file UI

### PR DESCRIPTION
**Pull request by oddsson**
# Improve case file UI

[Asana](https://app.asana.com/0/1199153462262248/1205942829380429/f)

## What

Improve case file UI depending on different states. Now the case file component acts as follows: 

- If a file is stored in RVG, it's blue and clickable.
- If a file was stored in RVG but has been removed, it's gray and not clickable.
- If a user tried to upload a file to Auður, but that failed, the file is red with a red retry button
- If a file is stored in Auður, it's either blue or gray (depending on whether it's stored in RVG or not) and has a blue checkmark icon.

There was also a rule on the ruling screen for district court users that stated that only assigned judges or registrars could open case files. This rule was removed. 

## Why

Clearer UI.

## Screenshots / Gifs

Screenshot 2024-01-04 at 13 13 22

Screenshot 2024-01-04 at 13 47 00

https://github.com/island-is/island.is/assets/3789875/cd2ee85d-b4bd-4579-882d-bfa9df07fefd

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by gudjong — APPROVED** [review]
Please remove console log in a few places.

File: apps/judicial-system/web/src/components/CaseFile/CaseFile.tsx [hunk 1]
```diff
@@ -1,56 +1,60 @@
...
+  onClick?: (fileId: string) => void
 }
 
 const CaseFile: React.FC<React.PropsWithChildren<Props>> = (props) => {
-  const { fileId, name, size, uploadedAt, canOpenFiles = true, onOpen } = props
-
-  const isValidUpdatedAtDate = isValid(new Date(uploadedAt))
+  const { name, size, color, id, icon, onClick } = props
 
+  console.log(icon)
```
Comment (oddsson) [inline apps/judicial-system/web/src/components/CaseFile/CaseFile.tsx:28]:
```suggestion
```

File: apps/judicial-system/web/src/utils/hooks/useCourtUpload/index.ts [hunk 2]
```diff
@@ -51,6 +51,8 @@ export const useCourtUpload = (
           status,
         }
 
+        console.log(files[fileIndexToUpdate])
```
Comment (oddsson) [inline apps/judicial-system/web/src/utils/hooks/useCourtUpload/index.ts:54]:
```suggestion
```

File: apps/judicial-system/web/src/utils/hooks/useCourtUpload/index.ts [hunk 3]
```diff
@@ -104,6 +105,7 @@ export const useCourtUpload = (
     if (files) {
       const xFiles = files as CaseFileWithStatus[]
       xFiles.forEach(async (file) => {
+        console.log(file)
```
Comment (oddsson) [inline apps/judicial-system/web/src/utils/hooks/useCourtUpload/index.ts:108]:
```suggestion
```

**Pull request merged**